pkg/cloudprovider/azure.go

```diff
@@ -302,6 +301,52 @@ func (a *Azure) GetNodeEgressIPConfiguration(node *corev1.Node, cloudPrivateIPCo
 	return []*NodeEgressIPConfiguration{config}, nil
 }
 
+// The consensus is to not add egress IP to public load balancer
+// backend pool regardless of the presence of an OutBoundRule.
+// During upgrade this function removes any egress IP added to
+// public load balancer backend pool previously.
+func (a *Azure) SyncLBBackend(ip net.IP, node *corev1.Node) error {
```

Comment:
have you tested this `SyncLBBackend`  function with so many EIPs already in place ? because AFAIK Azure APIs are so slow and not sure how it works when you want to sync already existing IPs. 
have you explored sync IPs belong to a node with single API call ? something similar to processing existing items (like [this](https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/go-controller/pkg/factory/factory.go#L1305-L1322)) before watching CloudPrivateIPConfig objects.

Reply:
let's wait for @kyrtapz opinion on this.